Ft works page

IrkaTyman opened the pull request.

IrkaTyman committed:
Начало работы над страницей работ + правки navbar

IrkaTyman committed:
1. Запрос,типы по skills
2. Запрос, типы по works/
3. Начальная страница работ(заглушка)

IrkaTyman committed:
--

IrkaTyman committed:
1. Документация
2. Компонент карточки работы (без ф-ции лайка)

IrkaTyman committed:
Ф-ция лайка работы

IrkaTyman committed:
Добавлены работы на страницу работ + заглушка

StalinidzeCorp committed:
PY-api-works-changes

PY-api-works-changes-2

IrkaTyman committed:
1. Временный адрес апи в конфиге
2. Компонент UserCard
3. Компонент User
4. Компонент PopularUsers
5. Модели ShortUser и ShortUserDto
6. Дописан сервис UsersService
7. Частичная верстка страницы работ

StalinidzeCorp committed:
PY-users-fix

IrkaTyman committed:
1. Стили
2. Панель поп пользователей

IrkaTyman committed:
1. Компоненты ErrorResult и EmptyResult
2. Стили Spin
3. Иконка FindIcon
4. Состояния загрузки и ошибки для SkillsSelect, UserFolderTabs
5. Стор useSkillsStore

IrkaTyman committed:
Merge remote-tracking branch 'origin/FT-works-page' into FT-works-page

# Conflicts:
#	frontend/src/components/SkillsSelect/SkillsSelect.module.scss
#	frontend/src/components/UserFolderTabs/UserFolderTabs.scss
#	frontend/src/components/UserFolderTabs/UserFolderTabs.tsx

IrkaTyman committed:
Правки стилей

IrkaTyman committed:
Правки:
1. Комментарии
2. Обработка состоний loading и error при загрузке works

IrkaTyman committed:
fetch -> axios in api

dmitrymaslov13 commented on frontend/src/components/Header/Header.tsx:
```diff
@@ -0,0 +1,46 @@
+
+// Компонент Хедер сайта
+const HeaderComponent: FC = () => {
+  useEffect(() => {
```
Вы уверены, что нам действительно нужно делать это так? 

Обычно хедер имеет фиксированную высоту, а основная страница имеет `overflow: auto;` и не будет никаких проблем с фиксацией

dmitrymaslov13 commented on frontend/src/components/Header/Header.tsx:
```diff
@@ -0,0 +1,46 @@
+  useEffect(() => {
+    /** Ф-ция фиксации шапки сайта. */
+    function onHeaderFixed() {
+      const header = document.getElementById('header');
```
FYI: Ты можешь использовать `useRef` и добавить атрибурт `ref` хедеру чтоб иметь ссылку на элемент. Использование нативного взаимодействия с DOM должно всегда вызывать сомнения

dmitrymaslov13 commented on frontend/src/components/Icons/UserAddIcon.tsx:
```diff
@@ -4,7 +4,7 @@ import { IconProps } from './types';
 // Иконка Пользователь с плюсом
 const UserAddIconComponent: FC<IconProps> = (props: IconProps) =>
-  <svg width={props.size} height={props.size} viewBox="0 0 22 20" fill="none" stroke="#000" xmlns="http://www.w3.org/2000/svg" {...props}>
+  <svg width={props.size} height={props.size} viewBox="0 0 22 20" fill="none" stroke="#000" xmlns="http://www.w3.org/2000/svg" {...props} className={'user-add_icon'}>
```
> ``
```suggestion
  <svg width={props.size} height={props.size} viewBox="0 0 22 20" fill="none" stroke="#000" xmlns="http://www.w3.org/2000/svg" {...props} className="user-add_icon">
```

dmitrymaslov13 commented on frontend/src/components/SkillsSelect/SkillsSelect.module.scss:
```diff
@@ -1,13 +1,15 @@
+  margin-bottom: var(--space-m);
+  background-color: #fff;
+  border: 1px solid var(--secondary-text-color);
+  border-radius: calc(var(--middle-text-fs) * var(--line-height) / 2 + var(--space-s));
```
Так как вы используете вложенность от SCSS, вы имеете возмжность вынести это значение в переменную этого стиля.
вы используете её на 11 строке

dmitrymaslov13 commented on frontend/src/components/WorkCard/WorkCard.tsx:
```diff
@@ -0,0 +1,86 @@
+import {useAuthStore} from '../../core/store/auth/store';
+
+import {WorksService} from '../../core/services/works-service';
+
```
Вы можете убрать пустые линии между импортами

dmitrymaslov13 commented on frontend/src/components/SkillsSelect/SkillsSelect.tsx:
```diff
@@ -0,0 +1,78 @@
+const SkillsSelectComponent: FC<Props> = props => {
+  /** Стор категорий. */
+  const {defaultSkills, isLoading, getSkills, error} = useSkillsStore();
+  /** Категории для выбора в панели. */
```
Для чего вы используете локальный стейт и обновляете его, если он итак подгружается из стора? 

вы можете напрямую использовать лоадинг получаемый из `useSkillsStore`, аналогично с ошибкой и со скиллами

dmitrymaslov13 commented on frontend/src/components/UserFolderTabs/UserFolderTabs.tsx:
```diff
@@ -1,22 +1,103 @@
+   * Ф-ция получения пользователей.
+   * @param selectedSkill - Выбранная категория.
+   */
+  const getUsers = async (selectedSkill: string) => {
```
Идея стора и сервисов нужна от части для того, чтоб в компоненте было минимум логики.

В идеале всё что вы должны делать в компоненте это хендлить метод стора, который будет обновлять всё остальное.

IrkaTyman committed:
Update frontend/src/components/Icons/UserAddIcon.tsx

Co-authored-by: Dmitry Maslov <90252321+dmitrymaslov13@users.noreply.github.com>

IrkaTyman committed:
Правки PR
1. Убраны строки между импортами
2. Хук для получения скиллов(с маппингом)
3. Хук на получение юзеров по скиллу
4. Стили хедера

IrkaTyman committed:
Merge remote-tracking branch 'origin/FT-works-page' into FT-works-page

StalinidzeCorp committed:
Merge branch 'main' into FT-works-page
